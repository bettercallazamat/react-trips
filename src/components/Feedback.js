import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { setFeedbackInactiveAction } from '../actions/feedback';
import '../assets/styles/modal.css';

const Feedback = ({ type, feedback }) => {
  const dispatch = useDispatch();
  const title = type === 'error' ? 'Error' : 'Success';

  const handleClick = () => {
    dispatch(setFeedbackInactiveAction());
  };

  return (
    <div className="Feedback modal">
      <div className="modal-content">
        <h3 className="title">{title}</h3>
        <p>{feedback}</p>
        <button className="cancel" type="button" onClick={handleClick}>OK</button>
      </div>
    </div>
  );
};

Feedback.propTypes = {
  type: PropTypes.string.isRequired,
  feedback: PropTypes.string.isRequired,
};

export default Feedback;
