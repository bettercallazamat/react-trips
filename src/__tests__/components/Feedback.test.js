import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Feedback from '../../components/Feedback';

describe('Feedback', () => {
  configure({ adapter: new Adapter() });
  const mockStore = configureStore([]);
  const store = mockStore({});
  store.dispatch = jest.fn();
  const type = 'error';
  const feedbackMessage = 'Error text';

  it('should match the snapshot', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Feedback feedback={feedbackMessage} type={type} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
