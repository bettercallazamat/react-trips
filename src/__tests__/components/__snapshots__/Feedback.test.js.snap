// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback should match the snapshot 1`] = `ShallowWrapper {}`;
